HW0/P4/P4.py

```diff
@@ -30,47 +30,101 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    
+    # load position and velocity variables from trajectory text file
+    x0, y0, z0, vx0, vy0, vz0 = np.loadtxt('/home/cs205/cs205-homework/HW0/P4/P4_trajectory.txt',delimiter=',',unpack=True)
+    
+    # plot position vectors wrt time on 3D axes
+    ax.plot(x0, y0, z0,'--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    
+    # create scaling matrix
+    R = np.matrix([[1/rx,0,0],[0,1/ry,0],[0,0,1/rz]])
+
+    # load measurements file as rows
+    m = np.loadtxt('/home/cs205/cs205-homework/HW0/P4/P4_measurements.txt',delimiter=',')
+    
+    # for each position given in measurements.txt, 
+    # matrix multiply scaling matrix R to position vector, 
+    # then typecast to a 1D array
+    adj = np.asarray([np.squeeze(np.asarray(np.dot(R,m[i]))) for i in np.arange(0,m.shape[0],1)])
+    
+    # transpose adj vector to get observed (x, y, z) coordinates
+    x_obs, y_obs, z_obs = adj.transpose()
+    
+    # plot observed positions
+    ax.plot(x_obs, y_obs, z_obs,'.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
-
-    # A = ?
-    # a = ?
-    # s = ?
-
-    # Initial conditions for s0
-    # Compute the rest of sk using Eq (1)
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    
+    # initialize A matrix and a vector
+    A = np.matrix([[1,0,0,dt,0,0],[0,1,0,0,dt,0],[0,0,1,0,0,dt],[0,0,0,1-c*dt,0,0],[0,0,0,0,1-c*dt,0],[0,0,0,0,0,1-c*dt]])
```

Comment:
I did the same thing. But my friend said this is actually not a very good style of coding